docs/flashbots-auction/miners/mev-geth-spec/v05.md

```diff
@@ -0,0 +1,216 @@
+
+A bundle `MUST` contain at least one transaction. There is no upper limit for the number of transactions in the bundle, however bundles that exceed the block gas limit will always be rejected. 
+
+A bundle `MAY` include `eligible coinbase payments`. Bundles that do not contain such payments may be discarded when their `bundle adjusted gas price` is compared with other bundles.
+
+The `maxTimestamp` value `MUST` be greater or equal the `minTimestamp` value.
+
+### Accepting bundles from the network
+
+#### JSON RPC
+
+Node `MUST` provide a way of exposing a JSON RPC endpoint accepting `eth_sendBundle` calls (specified [here](v02-rpc.mdx)). Such endpoint `SHOULD` only be accepting calls from a `relay` but there is no requirement to restrict it through the node source code as it can be done on the infrastructure level.
+
+Node `MUST` provide a way of exposing a JSON RPC endpoint accepting `eth_sendMegabundle` calls (specified [here](v04-rpc.mdx)). Such endpoint `SHOULD` only be accepting calls from a `relay` but there is no requirement to restrict it through the node source code as it can be done on the infrastructure level. For each relay only the last megabundle should be stored in memory.
+
+### Bundle eligibility
+
+Any bundle that is correctly constructed `MUST` have a `blockNumber` field set which specifies in which block it can be included. If the node has already progressed to a later block number then such bundle `MAY` be removed from memory.
+
+Each transaction in the bundle `MUST` have _priorityFeePerGas_ equal or greater than _block.BASEFEE_ + _1 GWei_ to be selected for the block.
+
+Any bundle that is correctly constructed `MAY` have a `minTimestamp` and/or a `maxTimestamp` field set. Default values for both of these fields are `0` and the meaning of `0` is that any block timestamp value is accepted. When these values are not `0`, then `block.timestamp` is compared with them. If the current `block.timestamp` is greater than the `maxTimestamp` then the bundle `MUST NOT` be included in the block and `MAY` be removed from memory. If the `block.timestamp` is less than `minTimestamp` then the bundle `MUST NOT` be included in the block and `SHOULD NOT` be removed from memory (it awaits future blocks).
+
+### Block construction
+
+In order to prevent starvation of less frequently used relays, incoming `MEV megabundles` should be given a priority value equal to the time in milliseconds since the previous `MEV megabundle` sent by the same relay and redirected to a pool of `MEV megabundle` block producers (called workers in MEV-Geth). Whenever the worker is idle and there are `MEV megabundles` available, a bundle with the highest priority should be picked for evaluation. Megabundles are never merged with other bundles and can only be combined with transactions from the mempool.
+
+When an `MEV megabundle` with higher value than previously arrives
+ * block building should be initated immediately if there is no block being built at the time.
```

Comment:
Also, "block building" is used here when "block construction" is used in other lines. I'd pick one.

Reply:
used construction, to separate from the relay side block building